nixos/modules/services/networking/godns.nix

```diff
@@ -0,0 +1,47 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.services.godns;
+
+  inherit (lib)
+    mkEnableOption
+    mkIf
+    mkOption
+    mkPackageOption
+    types
+    ;
+in
+{
+
+  options.services.godns = {
+    enable = mkEnableOption "GoDNS Service";
+
+    package = mkPackageOption pkgs "godns" { };
+
+    configPath = mkOption {
+      type = types.path;
+      description = "Path to the configuration file for godns.";
+      example = "/etc/godns/config.json";
+    };
```

Comment:
We should allow writing that with nixos options

Reply:
Got it, I would for now prefer to have this untyped, so just a generic Attribute set holding the config. Is that okay?

Reply:
Sorry @SuperSandro2000 do you mean a NixOS option specifying all available options in the configuration or a generic Attribute set or a string?

Reply:
See https://github.com/NixOS/rfcs/blob/master/rfcs/0042-config-option.md#detailed-design